testing/e2e/upgrade_testify_test.go

```diff
@@ -0,0 +1,86 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	tools "github.com/elastic/elastic-agent/testing/e2e/tools"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type UpgradeElasticAgent struct {
+	suite.Suite
+	clusterConfigPath string
+	clusterConfig     tools.ClusterConfig
+	client            *tools.Client
+	agentVersion      string
+}
+
+// Before suite
+func (suite *UpgradeElasticAgent) SetupSuite() {
+	suite.agentVersion = os.Getenv("AGENT_VERSION")
+	assert.NotEmpty(suite.T(), suite.agentVersion, "AGENT_VERSION is not set")
+
+	suite.clusterConfigPath = os.Getenv("CLUSTER_CONFIG_PATH")
+	if suite.clusterConfigPath == "" {
+		suite.clusterConfigPath = "../cluster-digest.yml"
+	}
+	var err error
+	suite.clusterConfig, err = tools.ReadConfig(suite.clusterConfigPath)
+	require.Nil(suite.T(), err, "Could not read cluster config")
+	suite.client, err = tools.NewClient(&suite.clusterConfig)
+	require.Nil(suite.T(), err, "Could not create Kibana client")
+
+	err = tools.DownloadElasticAgent(suite.agentVersion)
+	require.Nil(suite.T(), err, "Could not download Elastic Agent")
+}
+
+func (suite *UpgradeElasticAgent) TestUpgradeFleetManagedElasticAgent() {
+	policy, err := suite.client.CreatePolicy()
```

Comment:
This is an API request, it should probably have retries. The same applies to any function here that makes a network call.

Reply:
Added retries